src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -361,13 +363,16 @@ private Int64 GetRandomInt64(Int64 min, Int64 max)
             do
             {
                 // Randomly fill the buffer
-                this.Generator.NextBytes(buffer);
+                Generator.NextBytes(buffer);
                 randomUint64 = BitConverter.ToUInt64(buffer, 0);
                 // Get the last 'bitsToRepresentDiff' number of randon bits
                 randomUint64 &= mask;
             } while (uint64Diff <= randomUint64);
 
             double result = min * 1.0 + randomUint64 * 1.0;
+
+            Debug.Assert(min <= result, "lower bound <= random number");
+            Debug.Assert(result < max, "random number < upper bound");
```

Comment:
The same.

Reply:
For this case I'm not too sure.
At line 347 it will try to enforce maxValue as an excluding upper bound due to the Next(min, max) method but in this scenario I'm not sure if it will enforce maxValue as an exclusive upper bound.
In doubt I think it's best to at least stick with previous behavior in which the Assert came after the function call thus trying to enforce maxValue as an excluded upper bound in all cases. It's safer to have a script fail than to have it run with an unexpected value.
@PaulHigin ?

Reply:
I'd expect the same behavior (included or excluded) for both low and high boundaries.
Although we can guarantee this only for integers.

Reply:
In that case : 
- do you consider this resolved ?
- do you want different groups of debug.assert statements for each of the two scenarios I listed where each assert correctly follows what the generator should return as a value ?
- do you want a single debug.assert group which would be this one but with randomValue <= upper bound ?
- do you want @PaulHigin 's take on this ?
:smile:

Reply:
@pougetat Please remove all controversial changes from this PR  so that we can move forward

Reply:
ok ill put the debug.assert statements back where they were.